packages/blocks/embed/src/App.tsx

```diff
@@ -117,17 +118,75 @@ export const App: BlockComponent<AppProps> = ({
   // The default width the block takes up. Ideally it should be provided by the EA
   const blockWidthRef = useRef<number | null>(null);
 
+  const getBlockDefaultSize = useCallback(
```

Comment:
Extracted the block resizing calculator to its own function, using it to recalculate dimensions when:
1. User supplies an updated `initialHtml`, eg, when switching variants in the blockprotocol site demo.
2. User tries to embed a new URL, which generates a new value for `html`.